value/src/it/functional/src/test/java/com/google/auto/value/AutoValueTest.java

```diff
@@ -1595,4 +1595,16 @@ public void testCopyAnnotations() throws Exception {
         methodInSuperclass.getAnnotation(HairyAnnotation.class);
     assertEquals(annotationInSuperclass, annotationInSubclass);
   }
+
+  @AutoValue
+  abstract static class HProperty {
+    public abstract Object h();
+    public static HProperty create(Object h) {
+      return new AutoValue_AutoValueTest_HProperty(h);
+    }
+  }
+  public void testHProperty() throws Exception {
+    // h is the hashCode variable, if we have a property named h, we get "error: int cannot be dereferenced"
```

Comment:
Can you reword this comment so that it makes sense after the change? Something like:
```java
  // Checks that we can have a property called `h`. The generated hashCode() method has
  // a local variable of that name so there could be clashes if we aren't careful.
```